src/satellite/process.ts

```diff
@@ -713,17 +709,19 @@ export class SatelliteProcess implements Satellite {
           localChanges.changes,
           incomingChanges.changes
         )
-        const optype = mergeOpTypesAddWins(
-          localChanges.optype,
-          incomingChanges.optype
-        )
+        let optype
 
         /*
         console.log('merge_op_tag')
         console.log(localChanges)
         console.log(incomingChanges)
          */
         const clearTags = mergeOpTags(localChanges, incomingChanges)
+        if (clearTags.length == 0) {
```

Comment:
I still find it confusing that this is called clearTags because what determines the optype is the presence of shadow tag and why in the `apply()` we also set tags with clearTags. 

We need to test this functions in isolation please. Could you make a simpler version of some of your tests that just generates the entries, calls apply, and checks what is the vicinity of `'INSERT wins over DELETE and restored deleted values'